motoko/basic_dao/tests/proposal.test.sh

```diff
@@ -155,4 +155,3 @@ upgrade(DAO, wasm, init);
 call DAO.list_proposals();
 assert _[0].state == variant { succeeded };
 assert _[1].state == variant { rejected };
-assert _[2].state.failed ~= "has no update method 'transfer'";
```

Comment:
What is going wrong here? Is an ic-repl assumption broken because of a new version, or is it motoko?

Reply:
This proposal actually succeeded. Here is the result running on my local machine, you can see the 3rd element of the vec is a success.

This test has been failing on master for a long time though.

```
> call DAO.list_proposals()
vec {
  record {
    id = 2 : nat;
    votes_no = record { amount_e8s = 0 : nat };
    voters = opt record {
      principal "7qnzu-vvwir-5w3ab-stmps-fwy23-zwwhf-usqg7-msh5w-oip6b-2fhnk-rae";
      opt record {
        principal "5qoxe-3e76q-ysztp-wn34i-mxuwr-tbzdw-eaqz2-ewvc5-yxdzh-vrahl-3qe";
        null;
      };
    };
    state = variant { succeeded };
    timestamp = 1_714_124_769_695_341_449 : int;
    proposer = principal "ylt5z-dx2jk-wibsj-v2sd3-bvfqi-lt54y-uc2q3-trgkl-fmebf-2mdvx-yae";
    votes_yes = record { amount_e8s = 700 : nat };
    payload = record {
      method = "transfer";
      canister_id = principal "bnz7o-iuaaa-aaaaa-qaaaa-cai";
      message = blob "\44\49\44\4c\02\6c\02\fb\ca\01\68\d8\a3\8c\a8\0d\01\6c\01\b9\ef\93\80\08\7d\01\00\01\1d\fb\88\33\76\70\2e\d9\19\8e\98\05\22\4a\02\d8\ed\32\d4\71\08\3e\67\39\20\b5\30\8c\3f\02\64";
    };
  };
  record {
    id = 1 : nat;
    votes_no = record { amount_e8s = 700 : nat };
    voters = opt record {
      principal "7qnzu-vvwir-5w3ab-stmps-fwy23-zwwhf-usqg7-msh5w-oip6b-2fhnk-rae";
      opt record {
        principal "5qoxe-3e76q-ysztp-wn34i-mxuwr-tbzdw-eaqz2-ewvc5-yxdzh-vrahl-3qe";
        null;
      };
    };
    state = variant { rejected };
    timestamp = 1_714_124_768_432_163_377 : int;
    proposer = principal "ylt5z-dx2jk-wibsj-v2sd3-bvfqi-lt54y-uc2q3-trgkl-fmebf-2mdvx-yae";
    votes_yes = record { amount_e8s = 0 : nat };
    payload = record {
      method = "transfer";
      canister_id = principal "bnz7o-iuaaa-aaaaa-qaaaa-cai";
      message = blob "\44\49\44\4c\02\6c\02\fb\ca\01\68\d8\a3\8c\a8\0d\01\6c\01\b9\ef\93\80\08\7d\01\00\01\1d\fb\88\33\76\70\2e\d9\19\8e\98\05\22\4a\02\d8\ed\32\d4\71\08\3e\67\39\20\b5\30\8c\3f\02\64";
    };
  };
  record {
    id = 0 : nat;
    votes_no = record { amount_e8s = 400 : nat };
    voters = opt record {
      principal "5qoxe-3e76q-ysztp-wn34i-mxuwr-tbzdw-eaqz2-ewvc5-yxdzh-vrahl-3qe";
      opt record {
        principal "7qnzu-vvwir-5w3ab-stmps-fwy23-zwwhf-usqg7-msh5w-oip6b-2fhnk-rae";
        opt record {
          principal "ylt5z-dx2jk-wibsj-v2sd3-bvfqi-lt54y-uc2q3-trgkl-fmebf-2mdvx-yae";
          opt record {
            principal "mu2i7-rp3ra-zxm4b-o3emy-5gafe-jfafw-hnglk-hccb6-m44sb-njqrq-7qe";
            null;
          };
        };
      };
    };
    state = variant { succeeded };
    timestamp = 1_714_124_755_176_464_094 : int;
    proposer = principal "mu2i7-rp3ra-zxm4b-o3emy-5gafe-jfafw-hnglk-hccb6-m44sb-njqrq-7qe";
    votes_yes = record { amount_e8s = 500 : nat };
    payload = record {
      method = "update_system_params";
      canister_id = principal "bnz7o-iuaaa-aaaaa-qaaaa-cai";
      message = blob "\44\49\44\4c\03\6c\03\f2\c7\94\ae\03\01\83\93\c7\b9\06\01\d8\ed\fd\d9\06\01\6e\02\6c\01\b9\ef\93\80\08\7d\01\00\01\90\4e\00\00";
    };
  };
}
```

Reply:
Ok, thanks. Then we can merge.